### PR TITLE
auto binding design

### PR DESCRIPTION
**Pull request by wonderflow**
This is a supplemental proposal for [Resource Dependency in OAM](https://github.com/crossplane/oam-kubernetes-runtime/blob/master/design/resource-dependency.md)

### CODE REVIEW

**Comment by SevenYoung** [comment]
LGTM

File: design/auto-binding.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+In this case, Developer of web component don't need to specify name of secret, but he has to specify the key of secret.
+
+But if the developer don't know the secret name, how can he know the key? So I propose we should automatically bind
+secret name along with the key.
+
+## Proposal
+
+The overall idea is to add more binding ways into `dataInputs` field.
+
+For k8s `secret`, we can add `toSecretKeyRef` field into `dataInputs`:
```
Comment (zzxwill) [inline design/auto-binding.md:122]:
```suggestion
For k8s `secret`, we can add `toSecretKeyRef` field into `dataInputs` for component `webapp`:
```

Comment (zzxwill) [inline design/auto-binding.md:122]: At the first glance, I think it is for workload `db` and find it doesn't make sense. And then I find it should be appended for workload `webapp`. So just for clarification, I made the modification:)

File: design/auto-binding.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+```
+      dataInputs:
+        - valueFrom:
+            dataOutputName: alibaba-rdspostgresql-conn
+          toSecretKeyRef:
+            - secretKey: endpoint
+              path: spec.template.spec.containers[0].env[0].valueFrom
```
Comment (zzxwill) [inline design/auto-binding.md:130]: Use `fieldpath` instead?

**Comment by zzxwill** [comment]
LGTM

File: design/auto-binding.md (hunk 1 above)
Comment (wonderflow) [reply design/auto-binding.md]: good

File: design/auto-binding.md (hunk 2 above)
Comment (wonderflow) [reply design/auto-binding.md:130]: This is an object field, I'm not sure if it's confusing to use the same name.

File: design/auto-binding.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+```
+
+The oam runtime will automatically bind secret name and key to the workload.
+
+This means below fragment will be injected automatically.
+```yaml
+                  valueFrom:
+                    name: alibaba-rdspostgresql-conn
+                    secretKeyRef:
+                      key: endpoint
```
Comment (ryanzhang-oss) [inline design/auto-binding.md:197]: shouldn't this be
```yaml
           secretKeyRef:
                name: alibaba-rdspostgresql-conn
                key: endpoint
```
instead?

**Pull request closed without merging**